compiler/noirc_frontend/src/hir/mod.rs

```diff
@@ -274,7 +279,15 @@ impl Context<'_, '_> {
             // Check for name collisions of this generic
             let name = Rc::new(ident.0.contents.clone());
 
-            ResolvedGeneric { name, type_var, kind: generic.kind(), span }
+            let mut resolved_generic = ResolvedGeneric {
+                name,
+                type_var,
+                kind: Kind::Numeric(Box::new(Type::Error)),
+                span,
+            };
+            let kind = generic.kind().map_err(|typ| (resolved_generic.clone(), typ))?;
+            resolved_generic.kind = kind;
```

Comment:
I think this can be done more cleanly. Instead of returning a `Vec<Result<...>>` we can return either a `(Vec<ResolvedGeneric>, Vec<ResolverError>)` or just a `Vec<ResolvedGeneric>` and accept a `&mut Vec<ResolverError>` as an argument.

Then we don't need to initialize `resolved_generic` to an error kind (although it could also have been `Kind::Normal` since it'll be overwritten anyway). We could do:
```rs
let kind = generic.kind().unwrap_or_else(|error| {
    errors.push(error);
    Kind::Normal
});
let resolved_generic = ResolvedGeneric { name, type_var, kind, span };
```

Reply:
I went with accepting a mutable reference to the errors list as to further reduce any logic needed after a call to `resolve_generics`. Looks much better now.